src/core/basetypes/MCData.cpp

```diff
@@ -319,9 +319,7 @@ static bool isHintCharsetValid(String * hintCharset)
             return true;
         }
         
-        // If it's among the known charset, we want to try to detect it
-        // to validate that it's the correct charset.
-        if (!knownCharset->containsObject(hintCharset)) {
```

Comment:
Why does this `if` changes direction?

Reply:
This function validates hintCharset string. If knownCharset list contains hintCharset - hintCharset is valid.